src/opera_disp_tms/create_measurement_geotiff.py

```diff
@@ -125,34 +126,58 @@ def create_measurement_geotiff(measurement_type: str, frame_id: int, begin_date:
     product_name = create_geotiff_name(measurement_type, frame_id, begin_date, end_date)
     product_path = Path.cwd() / product_name
     data.rio.to_raster(product_path.name)
+
     return product_path
 
 
-def main():
-    """CLI entry point
-    Example:
-    create_measurement_geotiff displacement 11114 20140101 20260101
-    """
+def make_parser():
     parser = argparse.ArgumentParser(description='Create a short wavelength displacement or velocity geotiff')
+
+    def frame_type(frame):
+        frame = int(frame)
+
+        if not (1 <= frame <= 46986):
+            parser.error(f'Value {frame} must be between 1 and 46986')
+
+        return frame
```

Comment:
If you move this helper function to the top level, then you can `raise argparse.ArgumentTypeError(f'Value {frame} must be between 1 and 46986')` to get the correct error message formatting on the command-line.